### DISCUSSION TITLE
Can't Log In (Login Loop)

### DISCUSSION THREAD

**Discussion by jpark952** [Shelf-nu/shelf.nu#1140] [Q&A]
I am trying to get a set up using Docker, but i am unable to log in.
During the Sign up phase i am presented with a the OTP prompt, i recive the OTP email, but when i enter it i am redirected back to the OTP Promt window. 

looking in superbase users, the user is there and has been confirmed, If i attempt to sign in with this user, i get redirect back to the login screen. 

here is my docker command:

docker run -d \
  --name "shelf" \
  -e "DATABASE_URL=postgresql://postgres.gvokuvnfturextlctrqj:[password]@aws-0-eu-west-2.pooler.supabase.com:6543/postgres?pgbouncer=true" \
  -e "DIRECT_URL=postgresql://postgres.gvokuvnfturextlctrqj:[password]@aws-0-eu-west-2.pooler.supabase.com:5432/postgres" \
  -e 'SUPABASE_ANON_PUBLIC=[key]' \
  -e 'SUPABASE_SERVICE_ROLE=[key]' \
  -e 'SUPABASE_URL=https://gvokuvnfturextlctrqj.supabase.co' \
  -e 'SESSION_SECRET={secret}' \
  -e 'SERVER_URL=http://localhost:3000' \
  -e 'SMTP_HOST=smtp.gmail.com' \
  -e 'SMTP_USER=jpark736@gmail.com' \
  -e 'SMTP_FROM="Jack from shelf.nu" <jpark736@gmail.com' \
  -e 'SMTP_PWD={pwd}' \
  -e 'INVITE_TOKEN_SECRET={secret}' \
  -p 3000:8080 \
  --restart unless-stopped \
  ghcr.io/shelf-nu/shelf.nu:latest
  
  
  this is the log within docker when i attempt to log in
  
   <-- POST /login?_data=routes%2F_auth%2B%2Flogin
  --> POST /login?_data=routes%2F_auth%2B%2Flogin 204 540ms
  <-- GET /assets?_data=routes%2F_layout%2B%2F_layout
  --> GET /assets?_data=routes%2F_layout%2B%2F_layout 302 1ms
  <-- GET /assets?_data=routes%2F_layout%2B%2Fassets
  --> GET /assets?_data=routes%2F_layout%2B%2Fassets 302 1ms
  <-- GET /assets?_data=routes%2F_layout%2B%2Fassets._index
  --> GET /assets?_data=routes%2F_layout%2B%2Fassets._index 302 1ms
  <-- GET /login?redirectTo=/assets
  --> GET /login?redirectTo=/assets 200 16ms
  <-- GET /login
  --> GET /login 200 13ms
  <-- GET /login?redirectTo=/assets
  --> GET /login?redirectTo=/assets 200 9ms
  <-- GET /login
  --> GET /login 200 12ms

**Comment by alecmalloc** [comment]
Hey I'm facing a similar issue where when I try to load /assets it says "Oops, something went wrong". And takes me back to the log in page. With lots of 302 redirects on all GET requests.
In dev (npm dev) it runs perfectly fine. 
Let me know if you figure smt out.

**Comment by DonKoko** [comment]
Do you have the same error when running production build without docker? I just setup a fresh shelf install and ran the production build without issue so I am assuming it might be related to docker.

**Reply by alecmalloc** [reply]
Hey thanks for the speedy reply. I'll give it a try without docker this morning and let you know.

**Reply by alecmalloc** [reply]
Hey you're right it must be something related to docker. As I'm not too experienced with dockerizing apps yet, I just went with pm2 for now. Except for a couple bugs it works great now !
